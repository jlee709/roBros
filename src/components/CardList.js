import React from "react";
import Card from "./Card";

//maping each elm for prop pass
const CardList = ({ robros }) => {
  const cardArray = robros.map((user, i) => {
    return (
      <Card
        key={i}
        id={robros[i].id}
        name={robros[i].name}
        email={robros[i].email}
        alias={robros[i].alias}
      />
    );
  });
  return <div>{cardArray}</div>;
};

export default CardList;
