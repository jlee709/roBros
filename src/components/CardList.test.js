import { shallow } from "enzyme";
import React from "react";
import CardList from "./CardList";

it("expects a CardList to render", () => {
  const mockRobros = [
    {
      id: 1,
      name: "Allan turing",
      username: "machinima",
      email: "theend@isnear.com",
      alias: "theHunter"
    }
  ];
  expect(shallow(<CardList robros={mockRobros} />)).toMatchSnapshot();
});
