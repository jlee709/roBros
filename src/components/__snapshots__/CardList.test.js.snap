// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects a CardList to render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robros={
        Array [
            Object {
              "alias": "theHunter",
              "email": "theend@isnear.com",
              "id": 1,
              "name": "Allan turing",
              "username": "machinima",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          alias="theHunter"
          email="theend@isnear.com"
          id={1}
          name="Allan turing"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "alias": "theHunter",
          "email": "theend@isnear.com",
          "id": 1,
          "name": "Allan turing",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            alias="theHunter"
            email="theend@isnear.com"
            id={1}
            name="Allan turing"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "alias": "theHunter",
            "email": "theend@isnear.com",
            "id": 1,
            "name": "Allan turing",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
